File: crate/fixtures/documents/patterns.cs
using System.Text.RegularExpressions;

class Validate {
	// Verbatim string: the source as written
	static readonly Regex Nested = new Regex(@"(a+)+");
	// Quoted string: escapes a level
	static readonly Regex Escaped = new Regex("\\d{4}");
	// The static form takes the subject first and the pattern second
	static bool Check(string s) => Regex.IsMatch(s, @"(a|ab)+");
	static string Strip(string s) => Regex.Replace(s, "\\s+", "");
}

File: crate/fixtures/documents/patterns.java
import java.util.regex.Pattern;

class Validate {
	static final Pattern NESTED = Pattern.compile("(a+)+");
	static final Pattern ESCAPED = Pattern.compile("\\d{4}-\\d{2}");
	static final Pattern QUALIFIED = java.util.regex.Pattern.compile("^[A-Z][a-z]+$");
	static boolean check(String s) {
		return Pattern.matches("(a|ab)+", s);
	}
}
